Fix Metricbeat/vsphere - Error "datastore '*' not found" #4879

amandahla opened the pull request:
We needed to change the way we use govmomi for collect metrics and now it works with versions 6.0/6.5. 

We have to remove 'datacenter' field also. 

Config.yml was corrected.

elasticmachine commented:
Can one of the admins verify this patch?

elasticmachine commented:
Since this is a community submitted pull request, a Jenkins build has not been kicked off automatically on build-eu-00. Can an Elastic organization member please verify the contents of this patch and then kick off a build manually?

exekias commented:
Could you please add a CHANGELOG entry?

Also, do you know if the new client version may help on this? https://github.com/elastic/beats/issues/4673, have you experienced any issue like that?

tsg commented:
jenkins, test it

amandahla commented:
@exekias  I updated the govmomi (0.15.0 now) and changed the way to collect. For the issue #4673 , I think that is possible to use something like this https://github.com/vmware/govmomi/issues/224 
I'm not sure how it will work :-|

exekias commented:
ok to test.

@amandahla is this ready for a new review?

amandahla commented:
@exekias Yes. Just had that issue with the flag 'version'.

exekias commented:
code looks good to me, there are some issues building it for other systems, do you have any idea of what's going on there? https://travis-ci.org/elastic/beats/jobs/265294456

exekias commented:
I'll have a look into the issue

amandahla commented on metricbeat/module/vsphere/host/host.go:
```diff
@@ -48,87 +45,78 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	}
+
+	for _, hs := range hst {
+		totalCpu := int64(hs.Summary.Hardware.CpuMhz) * int64(hs.Summary.Hardware.NumCpuCores)
```
Solved.

amandahla commented on metricbeat/module/vsphere/host/host.go:
```diff
@@ -48,87 +45,78 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+
+	for _, hs := range hst {
+		totalCpu := int64(hs.Summary.Hardware.CpuMhz) * int64(hs.Summary.Hardware.NumCpuCores)
+		freeCpu := int64(totalCpu) - int64(hs.Summary.QuickStats.OverallCpuUsage)
```
Solved.

ruflin commented:
@amandahla @exekias What is the status here? 

@amandahla Any chance you could rebase this on master and then we could have a look again.

exekias commented:
I had a look at the issue but couldn't get it working, looks like the client lacks support for many architectures  :/

amandahla commented:
Using session we were getting errors with architectures but now opening a new client on every fetch seems ok. It's the same way like before but instead doing a new connection only once, we do it for every fetch.

ruflin commented:
@amandahla Could you rebase this one again for the changelog?

ruflin commented:
@amandahla I did 2 commits myself to see if I can fix the CHANGELOG issue and getting this in.

amandahla commented:
@ruflin Thanks!

exekias commented:
Awesome contribution @amandahla, I'm glad this got finally in!! :tada: :tada:

amandahla commented:
@exekias Thank you, I'm very happy with that! :smile:
